packages/flutter/lib/src/material/radio.dart

```diff
@@ -33,18 +37,306 @@ enum _RadioType { material, adaptive }
 const double _kOuterRadius = 8.0;
 const double _kInnerRadius = 4.5;
 
+/// A group for [Radio]s.
+///
+/// This widget treats all [Radio]s in the sub tree with the same type T as a
+/// group.
+///
+/// This widget handles the group value for the [Radio]s in the subtree with the
+/// same value type. To use this widget, the [Radio.groupValue] must leave as
+/// null. Otherwise, an assertion error is thrown.
+///
+/// This widget internal builds a [RadioGroupPolicy] to provide an accessible
+/// way to interact with the [Radio]s in this group.
+///
+/// Consider using [RadioGroupPolicy] if one requires custom logic when
+/// selecting radio and thus needs to manually supply group value to each
+/// [Radio].
+///
+/// {@tool dartpad}
+/// Here is an example of RadioGroup widget.
+///
+/// Try using tab, arrow keys, and space to see how the widget responds.
+///
+/// ** See code in examples/api/lib/material/radio/radio_group.0.dart **
+/// {@end-tool}
+class RadioGroup<T> extends StatefulWidget {
+  /// creates a radio group
+  ///
+  /// The `initialGroupValue` set the initial selection on subtree [Radio] with
+  /// the same [Radio.value].
+  ///
+  /// The `onChanged` is called when the selection has changed in the subtree
+  /// [Radio]s.
+  const RadioGroup({super.key, this.initialGroupValue, this.onChanged, required this.child})
+    : super();
+
+  /// The initial selection on the [Radio] of the matching [Radio.value].
+  final T? initialGroupValue;
```

Comment:
This pattern makes me hesitate, even though I know we've done it elsewhere in the framework. I worry that there are cases where it gets confusing whether an initialValue should be applied or not, like rebuilding the widget, changing the initialValue after user interaction, moving the widget, etc.